src/threat_model/rust_client_threat_model.md

```diff
@@ -0,0 +1,134 @@
+their own. The identity will be used by the service to partition the assets it stores for clients
+into a namespace for each one.
+
+**Confidentiality** : if known (the authentication token), an attacker could impersonate a specific
+application and execute operations in their name.
+
+### Client’s data- AS2
+
+Data sent through the client as part of a request to the service.
+
+**Confidentiality** : some of the client’s data could be confidential (example: a buffer to
+encrypt).
+
+**Integrity** : the data should not be modifiable by attackers. The client should be certain that it
+is interacting with the legitimate Parsec service.
+
+### Configuration data- AS3
+
+Configuration data passed by the client application to the library. There is some overlap between
+the configuration data and the application identity, as the identity is stored within the client
+library and could thus be considered part of its configuration data.
+
+**Confidentiality** : if known (the authentication token), an attacker could impersonate a specific
+application and execute operations in their name.
+
+**Integrity** : the data should only be modified by the client application as doing otherwise might
+prevent the client application from accessing Parsec functionality.
+
+### Client’s cryptographic keys - AS4
+
+Data stored in the configuration file, read at each reloading of Parsec to instantiate components.
+
+**Confidentiality** : (private) key material is sensitive data and should not be exposed.
+
+**Integrity** : neither public nor private key data should be modifiable by attackers.
+
+**Availability** : the client must always be able to use their cryptographic material.
+
+## Attackers
+
+Each dataflow is analysed from an attacker’s perspective using STRIDE method. Nothing is supposed
+on the different components.
+
+In the following tables are present the type of each possible threat, its description, its
+mitigation status and the assets impacted. A threat can be unmitigated (U), mitigated (M) or
+operational (O). The assumptions context applies for all threats but when one of them is
+particularly relevant, it will be noted with ASUM.
+
+### Attacker “Request to Service” - A1
+
+This attacker uses the inter-process endpoint, presumably created by the service, to interfere with
+the communication between client and service.
+
+|   | Description                                                                                                                                       | Mitigation | Assets        |
+|---|---------------------------------------------------------------------------------------------------------------------------------------------------|------------|---------------|
+| S | An attacker impersonates the service by establishing an IPC endpoint that spoofs the one used by Parsec to exfiltrate any data found in requests. | O-0, U-1   | AS1, AS2, AS4 |
```

Comment:
I think as well as the faculty of spoofing the service to read requests, a spoofed service can also return maliciously crafted data (like signing data with the attacker's key instead of Parsec's).